Improve smiley editor

BrickOzp opened the pull request:
* Default "Same image for all sets" to checked, when adding
new smiley.
* Add help text about "Same image for all sets".
* Make it possible to upload new images when editing existing
smileys.

Fixes #5368

Signed-off-by: Oscar Rydhé oscar.rydhe@gmail.com

SGA-oscar-rydhe committed:
Improve smiley editor

* Default "Same image for all sets" to checked, when adding
new smiley.
* Add help text about "Same image for all sets".
* Make it possible to upload new images when editing existing
smileys.

Fixes #5368

Signed-off-by: Oscar Rydhé oscar.rydhe@gmail.com

live627 commented on Sources/ManageSmileys.php:
```diff
@@ -1011,6 +1013,58 @@ function EditSmileys()
+				$writeErrors = array();
+				foreach ($_FILES['smiley_upload']['name'] as $set => $name)
+				{
+					if (!empty($name))
+						if (!is_writable($context['smileys_dir'] . '/' . $set))
```
Can these two conditions be combined?

BrickOzp replied:
Sure can

live627 commented on Sources/ManageSmileys.php:
```diff
@@ -1011,6 +1013,58 @@ function EditSmileys()
+				}
+
+				if (!empty($writeErrors))
+					fatal_lang_error('smileys_upload_error_notwritable', true, array(implode(', ', $writeErrors)));
```
General question to whomever: should these errors be logged? (imo they shouldn't, but there may be a reason here that I missed)

BrickOzp replied:
I did also wonder about that. This seems to be used as error handling in many places in the admin section. Is there another way that will just show the error to the user?
The code is mostly copied from AddSmiley()

SGA-oscar-rydhe committed:
Combine if cases into one

Signed-off-by: Oscar Rydhé oscar.rydhe@gmail.com

SGA-oscar-rydhe committed:
Do not log errors in ManageSmileys

These errors are realted to inputs from forms
when adding or modifying smileys. There is no need
to add these errors to the log.

Signed-off-by: Oscar Rydhé oscar.rydhe@gmail.com
